### PR TITLE
py-pygit2, libgit2: add new versions

### PR DESCRIPTION
**Pull request by ketsubouchi**
Add latest versions and dependencies.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pygit2/package.py [hunk 1]
```diff
@@ -14,15 +14,27 @@ class PyPygit2(PythonPackage):
     homepage = "http://www.pygit2.org/"
     pypi = "pygit2/pygit2-0.24.1.tar.gz"
 
+    version('1.4.0', sha256='cbeb38ab1df9b5d8896548a11e63aae8a064763ab5f1eabe4475e6b8a78ee1c8')
+    version('1.3.0', sha256='0be93f6a8d7cbf0cc79ae2f0afb1993fc055fc0018c27e2bd01ba143e51d4452')
+    version('0.28.2', sha256='4d8c3fbbf2e5793a9984681a94e6ac2f1bc91a92cbac762dbdfbea296b917f86')
     version('0.24.1', sha256='4d1d0196b38d6012faf0a7c45e235c208315672b6035da504566c605ba494064')
 
     extends('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pygit2/package.py]: This line isn't needed, it's already added by the base class

File: var/spack/repos/builtin/packages/py-pygit2/package.py [hunk 2]
```diff
@@ -14,15 +14,27 @@ class PyPygit2(PythonPackage):
...
+    depends_on('libgit2@0.99:1.0.99', when='@1.1:1.1.99')
+    depends_on('libgit2@0.28:0.28.99', when='@0.28:1.0.99')
+    depends_on('libgit2@0.24:0.27.99', when='@0.24:0.27.99')
+    depends_on('python@3.6:', when='@1.4.0:')
+    depends_on('python@3.6:3.8.99', when='@1.2:1.3.99')
+    depends_on('python@3.5:3.8.99', when='@1.0:1.1.99')
+    depends_on('python@2.7:3.87.99', when='@0.28:0.99.99')
     depends_on('py-six', type=('build', 'run'))
     depends_on('py-cffi', type=('build', 'run'))
+    depends_on('py-cached-property', when='@1.1.0:', type=('run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pygit2/package.py]:
```suggestion
    depends_on('py-cached-property', when='@1.1.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pygit2/package.py [hunk 3]
```diff
@@ -14,15 +14,27 @@ class PyPygit2(PythonPackage):
...
+    depends_on('libgit2@1.0:1.0.99', when='@1.2:1.3.99')
+    depends_on('libgit2@0.99:1.0.99', when='@1.1:1.1.99')
+    depends_on('libgit2@0.28:0.28.99', when='@0.28:1.0.99')
+    depends_on('libgit2@0.24:0.27.99', when='@0.24:0.27.99')
+    depends_on('python@3.6:', when='@1.4.0:')
+    depends_on('python@3.6:3.8.99', when='@1.2:1.3.99')
+    depends_on('python@3.5:3.8.99', when='@1.0:1.1.99')
+    depends_on('python@2.7:3.87.99', when='@0.28:0.99.99')
     depends_on('py-six', type=('build', 'run'))
     depends_on('py-cffi', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pygit2/package.py]:
```suggestion
    depends_on('py-cffi@1.4.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pygit2/package.py [hunk 4]
```diff
@@ -14,15 +14,27 @@ class PyPygit2(PythonPackage):
...
-    depends_on('libgit2@0.24:', when='@0.24:')
+    depends_on('libgit2@1.1:1.1.99', when='@1.4:')
+    depends_on('libgit2@1.0:1.0.99', when='@1.2:1.3.99')
+    depends_on('libgit2@0.99:1.0.99', when='@1.1:1.1.99')
+    depends_on('libgit2@0.28:0.28.99', when='@0.28:1.0.99')
+    depends_on('libgit2@0.24:0.27.99', when='@0.24:0.27.99')
+    depends_on('python@3.6:', when='@1.4.0:')
+    depends_on('python@3.6:3.8.99', when='@1.2:1.3.99')
+    depends_on('python@3.5:3.8.99', when='@1.0:1.1.99')
+    depends_on('python@2.7:3.87.99', when='@0.28:0.99.99')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pygit2/package.py]:
```suggestion
    depends_on('python@2.7:3.8.99', when='@0.28:0.99.99')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
@AndrewGaspar can you review?

**Comment by AndrewGaspar** [comment]
libgit2 looks fine - my checksum agrees.

**Review by AndrewGaspar — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-pygit2/package.py (hunk 4 above)
Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/py-pygit2/package.py]: 'python@2.7:3.7.99' is right for when='@0.28:0.99.99'

Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/py-pygit2/package.py]: See b7d0022

File: var/spack/repos/builtin/packages/py-pygit2/package.py (hunk 1 above)
Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/py-pygit2/package.py]: Version 0.24.1 is too old. Build of libgit2 0.24.1 is unable.
Version 0.28.2 needs old Python.
So  version 1.3.0 seems better as "typical" version.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pygit2/package.py]: I'm talking about the `extends('python')` line

Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/py-pygit2/package.py]: Remove the line. Please see c4dfecf .

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**